### PR TITLE
Add content to Users tab on admin panel

### LINKED ISSUE #333: Add content to Users tab to admin panel
Access for this tab should have only admin

image

### PR DESCRIPTION
**Pull request by mhevyk**
I tried to implement this logic before actual endpoint implementation, so i guess we can finally sync this tab after completion this task on backend

image

image

image

image

image

### CODE REVIEW

File: code/src/constants/common.ts [hunk 1]
```diff
@@ -17,3 +17,8 @@ export const ROLES = {
   SHOP_MANAGER: "ROLE_MANAGER",
   ADMIN: "ROLE_ADMIN"
 } as const;
+
+export const USER_STATUSES = {
+  ACTIVE: "ACTIVE",
+  DEACTIVATED: "DEACTIVATED"
+} as const;
```
Comment (dmtrth25) [inline code/src/constants/common.ts:24]: In general what do you think about `enum` approach ?
I thought about it but didn't catch why `const` obj with properties better ?
What do you think ?

File: code/src/containers/tables/users-table/components/users-table-body/UsersTableBody.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      <AppTableCell>
+        <AppTypography variant="caption">{user.email}</AppTypography>
+      </AppTableCell>
+      <AppTableCell>{roleBadge}</AppTableCell>
+      <AppTableCell>
+        <AppTypography variant="caption">
+          {formatDate(user.createdAt)}
+        </AppTypography>
+      </AppTableCell>
+      <AppTableCell>{statusBadge}</AppTableCell>
```
Comment (dmtrth25) [inline code/src/containers/tables/users-table/components/users-table-body/UsersTableBody.tsx]: what do you think about destruction `user` properties ? `status` `role` above too.
And also render some variable instead `{user.firstName} {user.lastName}`

File: code/src/constants/common.ts (hunk 1 above)
Comment (mhevyk) [reply code/src/constants/common.ts:24]: Discussed

File: code/src/containers/tables/users-table/components/users-table-body/UsersTableBody.tsx (hunk 2 above)
Comment (mhevyk) [reply code/src/containers/tables/users-table/components/users-table-body/UsersTableBody.tsx]: Fixed

**Review by dmtrth25 — APPROVED** [review]

**Review by OliaHrytsak — APPROVED** [review]

**Comment by D1nviz** [comment]
I suggest adding width property to these badges. Make them similar
image

File: code/src/containers/tables/users-table/components/users-table-body/UsersTableBody.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+import AppBadge from "@/components/app-badge/AppBadge";
+import { AppTableCell } from "@/components/app-table/components";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import formatDate from "@/utils/format-date/formatDate";
+
+const UsersTableBody = ({ user }: UsersTableBodyProps) => {
+  const { firstName, lastName, role, status, createdAt, email } = user;
+  const fullName = `${firstName} ${lastName}`;
```
Comment (D1nviz) [inline code/src/containers/tables/users-table/components/users-table-body/UsersTableBody.tsx]: A full name is also used in another place. Maybe create function for contact first and last names? You can create the separate issue for that

**Review by ValKoDim — APPROVED** [review]

**Review by heshvab — APPROVED** [review]

**Comment by mhevyk** [comment]
> I suggest adding width property to these badges. Make them similar image

Thanks a lot! I did as you suggested, now the width of badge is calculated based on longest badge translation

File: code/src/containers/tables/users-table/components/users-table-body/UsersTableBody.tsx (hunk 3 above)
Comment (mhevyk) [reply code/src/containers/tables/users-table/components/users-table-body/UsersTableBody.tsx]: I created util and reused it

**Review by D1nviz — APPROVED** [review]

**Pull request merged**